libraries/AP_DDS/AP_DDS_Client.cpp

```diff
@@ -410,21 +410,27 @@ void AP_DDS_Client::on_topic (uxrSession* uxr_session, uxrObjectId object_id, ui
     1) Store the read contents into the ucdr buffer
     2) Deserialize the said contents into the topic instance
     */
-    sensor_msgs_msg_Joy* topic = nullptr;
-    const bool success = sensor_msgs_msg_Joy_deserialize_topic(ub, topic);
 
-    if (success == false || topic == nullptr) {
-        return;
-    }
+    switch (object_id.id) {
+    case topics[to_underlying(TopicIndex::JOY_SUB)].dr_id.id:
```

Comment:
Not that I know of, since we don't have access to the STL templating library, there is no static analysis that can be done. Are you ok with enforcing it in PR's?

Reply:
This is going to be much easier to maintain and manage concurrent PRs adding topcis - we'll just have to remember to check the order carefully.